packages/nimble-components/src/rich-text/models/markdown-parser.ts

```diff
@@ -25,6 +26,10 @@ export class RichTextMarkdownParser {
      */
     private static mentionConfigs?: MarkdownParserMentionConfiguration[];
 
+    public static getMentionedHrefs(): string[] {
```

Comment:
I may be misreading something, but isn't this a case of a non-static method on the component calling a static method on the parser? How does this work? Doesn't the parser need context like the markdown and the mention config (similar to what's being passed to `parseMarkdownToDOM?` below)?

edit: I see that the parser is caching the mentioned hrefs from the most recent markdown that was parsed. And then I guess it's up to the caller component to retrieve those cached mentions before someone else calls parse. This seems really brittle. If there are multiple viewer components on the same page and the client waits until all of the viewers have been parsed before calling `getMentionedHrefs`, I think they'll all return the mentions from whichever viewer was parsed last.

Can we come up with a different pattern (e.g. returning the mentioned users from the parse method)?

Reply:
As we discussed in the call, let's update `parseMarkdownToDOM()` to return an object that contains both the document fragment and the list of mentioned users. You'd follow similar patterns for storing/clearing the static data as we use already in this class. Then the viewer can cache the result, return it from its `mentionedHrefs` accessor, and update the cached result every time the markdown is set.

Reply:
Updated the parser logic. Now `parseMarkdownToDOM()` will return object having `fragment` and `mentionedHrefs`, Will be caching that in viewer and returning it on `getMentionedHrefs()` method.